File: Pembuat Pola Segitiga/Pembuat-Pola-Segitiga.cpp
#include <iostream>

void segitiga_siku_siku_pola1 ();
void segitiga_siku_siku_pola1_terbalik ();
void segitiga_siku_siku_pola2 ();
void segitiga_siku_siku_pola2_terbalik ();
void segitiga_sama_kaki ();

int main ()
{
    unsigned short pilihan;

    std::cout << "========== Pembuat Pola Segitiga ==========\n";
    std::cout << "1. Segitiga siku-siku pola 1\n";
    std::cout << "2. Segitiga siku-siku pola 1 (terbalik)\n";
    std::cout << "3. Segitiga siku-siku pola 2\n";
    std::cout << "4. Segitiga siku-siku pola 2 (terbalik)\n";
    std::cout << "5. Segitiga sama kaki\n";
    std::cout << "Pilih salah satu: ";
    std::cin >> pilihan;

    switch (pilihan)
    {
        case 1:
            segitiga_siku_siku_pola1 ();
            break;
        case 2:
            segitiga_siku_siku_pola1_terbalik ();
            break;
        case 3:
            segitiga_siku_siku_pola2 ();
            break;
        case 4:
            segitiga_siku_siku_pola2_terbalik ();
            break;
        case 5:
            segitiga_sama_kaki ();
            break;
        default:
            std::cout << "Input error\n";
            break;
    }

    return 0;
}

void segitiga_siku_siku_pola1 ()
{
    unsigned short tinggi = 0;

    std::cout << "Masukkan tinggi segitiga: ";
    std::cin >> tinggi;

    for (unsigned short i = 1; i <= tinggi; i++)
    {
        for (unsigned short j = 1; j <= i; j++)
        {
            std::cout << "*";
        }
        std::cout << "\n";
    }
}

void segitiga_siku_siku_pola1_terbalik ()
{
    unsigned short tinggi = 0;

    std::cout << "Masukkan tinggi segitiga: ";
    std::cin >> tinggi;

    for (unsigned short i = 0; i <= tinggi; i++)
    {
        for (unsigned short j = 1; j <= tinggi - i; j++)
        {
            std::cout << "*";
        }
        std::cout << "\n";
        
        if (i == tinggi - 1)
        {
            break;
        }

    }
}

void segitiga_siku_siku_pola2 ()
{
    unsigned short tinggi = 0;

    std::cout << "Masukkan tinggi segitiga: ";
    std::cin >> tinggi;

    for (unsigned short i = 1; i <= tinggi; i++)
    {
        for (unsigned short j = 1; j <= tinggi - i; j++)
        {
            std::cout << " ";
        }
        for (unsigned short k = 1; k <= i; k++)
        {
            std::cout << "*";
        }
        std::cout << "\n";
    }
}

void segitiga_siku_siku_pola2_terbalik ()
{
    unsigned short tinggi = 0;

    std::cout << "Masukkan tinggi segitiga: ";
    std::cin >> tinggi;

    for (unsigned short i = 0; i <= tinggi; i++)
    {
        for (unsigned short j = 1; j <= i; j++)
        {
            std::cout << " ";
        }
        for (unsigned short k = 1; k <= tinggi - i; k++)
        {
            std::cout << "*";
        }
        std::cout << "\n";

        if (i == tinggi - 1)
        {
            break;
        }
    }
}

void segitiga_sama_kaki ()
{
    unsigned short tinggi = 0;

    std::cout << "Masukkan tinggi segitiga: ";
    std::cin >> tinggi;

    for (unsigned short i = 1; i <= tinggi; i++)
    {
        for (unsigned short j = 1; j <= tinggi - i; j++)
        {
            std::cout << " ";
        }
        for (unsigned short k = 1; k <= 2 * i - 1; k++)
        {
            std::cout << "*";
        }
        std::cout << "\n";
    }
}

File: Pembuat Pola Segitiga/Pembuat-Pola-Segitiga.py
def segitiga_siku_siku_pola1 ():
    tinggi = int (input ("Masukkan tinggi segitiga: "))
    
    for i in range (1, tinggi + 1):
        for j in range (i):
            print ("*", end = "")
        print ()

def segitiga_siku_siku_pola1_terbalik ():
    tinggi = int (input ("Masukkan tinggi segitiga: "))

    for i in range (tinggi, 0, -1):
        for j in range (i):
            print ("*", end = "")
        print ()

def segitiga_siku_siku_pola2 ():
    tinggi = int (input ("Masukkan tinggi segitiga: "))
    
    for i in range (1, tinggi + 1):
        for j in range (tinggi - i):
            print (" ", end = "")
        for k in range (i):
            print ("*", end = "")
        print ()

def segitiga_siku_siku_pola2_terbalik ():
    tinggi = int (input ("Masukkan tinggi segitiga: "))
    
    for i in range (tinggi + 1):
        for j in range (i):
            print (" ", end = "")
        for k in range (tinggi - i):
            print ("*", end="")
        print ()

def segitiga_sama_kaki ():
    tinggi = int (input ("Masukkan tinggi segitiga: "))
    
    for i in range (1, tinggi + 1):
        for j in range (tinggi - i):
            print (" ", end = "")
        for k in range (2 * i - 1):
            print ("*", end = "")
        print ()

print ("========== Pembuat Pola Segitiga ==========")
print ("1. Segitiga siku-siku pola 1")
print ("2. Segitiga siku-siku pola 1 (terbalik)")
print ("3. Segitiga siku-siku pola 2")
print ("4. Segitiga siku-siku pola 2 (terbalik)")
print ("5. Segitiga sama kaki")
pilihan = int (input ("Pilih salah satu: "))

if pilihan == 1:
    segitiga_siku_siku_pola1 ()
elif pilihan == 2:
    segitiga_siku_siku_pola1_terbalik ()
elif pilihan == 3:
    segitiga_siku_siku_pola2 ()
elif pilihan == 4:
    segitiga_siku_siku_pola2_terbalik ()
elif pilihan == 5:
    segitiga_sama_kaki ()
else:
    print ("Input error")
